draft-ounsworth-cfrg-kem-combiners.mkd

```diff
@@ -396,42 +391,49 @@ This is a non-comprehensive list, further information can be found in paragraph
-1. `KDF = SHA3-512`, with `hashSize = 512 bit`.
+| Instantiation number | KDF construction | H(x)     | hashSize | outputBits |
+| ---                  | ---              | ---      | ---      | ---        |
+| 1                    | {#sec-kmac-kdf}  | KMAC128  | 128 bit  | Variable   |
+| 2                    | {#sec-kmac-kdf}  | KMAC256  | 256 bit  | Variable   |
+| 3                    | {#sec-hash-kdf}  | SHA3-256 | 256 bit  | 256 bit    |
+| 4                    | {#sec-hash-kdf}  | SHA3-512 | 512 bit  | 512 bit    |
+{: #tab-kdfs title="KDF instantiations"}
 
 As justified in the security considerations, we recommend only Keccak-based instantiations because assuming there are no weaknesses found in the Keccak permutation, it behaves as a split-key PRF that can be keyed by any input `k_i`.
 SHAKE is also not included in the list as it is not allowed by {{SP800-56A}} section 7, and does not provide any implementation advantage over KMAC.
 
 KMAC constructions are RECOMMENDED over SHA-3, as KMAC offers a simple cSHAKE-based construction, with the advantage of returning an unrelated output when requesting a different `outputBits` KEK length.
 
-## KMAC based construction
-Options 1 and 2 are KMAC-based, as specified in NIST SP 800-185 {{SP800-185}}.
-To instantiate the function:
+## KMAC based construction {#sec-kmac-kdf}
+
+This section defines a `KDF(Z, OtherInput, outputBits)` as per {{SP800-56C}} section 4.1 Option 3 using the KMAC primitive as specified in NIST SP 800-185 {{SP800-185}}.
+To instantiate `KMAC#`:
 
-* The context `S` MUST be the utf-8 string "KDF".
-* The key `K` MUST be a context-specific string of at least `hashSize` bits, and it MAY be used as an additional option to perform context separation, in scenarios where `fixedInfo` is not sufficient.
-* The parameter `counter` MUST be the fixed value `0x00000001`, corresponding to a 32-bit binary big-endian representation of the number zero.
-    * This aligns with {{SP800-56C}} where `counter` is initialized to `0x00000000` is Section 4.1 Step 3, but is first incremented in 6.1 before being used in a hash in 6.2.
+KMAC is defined in NIST SP 800-185 [SP800-185]. The `KMAC#(K, X, L, S)` parameters MUST be instantiated as follows:
+
+* `KMAC#` : either `KMAC128` or `KMAC256` is per {{tab-kdfs}}.
+* `K`: a context-specific string of at least `hashSize` bits, and it MAY be used as an additional option to perform context separation, in scenarios where `OtherInput` is not sufficient.
+* `X`: the message input to `KDF()`, as defined above.
+* `L`: integer representation of `outputBits`.
+* `S`: as specified in {{SP800-56C}}, it is the 8-bit ASCII string string "KDF".
+
+
+Since we are setting `L = outputBits`, Step 1 of the process in {{SP800-56C}} spction 4.1 will always be `reps = ciel(L / H_outputBits) = 1`, and since the input strings defined in {{sec-kem-combiner}} are fixed-length and and well below the max input size of KMAC, we can also skip step 4.
```

Comment:
Re "well below the max input size of KMAC", I don't think there is a max input size; SP800-56Cr2 says "if Option
3 is chosen for the implementation of H, then there is no specification-imposed upper bound
on max_H_inputBits"

Reply:
Good catch.